barretenberg/cpp/src/barretenberg/vm2/simulation/alu.test.cpp

```diff
@@ -80,14 +76,13 @@ TEST(AvmSimulationAluTest, NegativeAddTag)
 TEST(AvmSimulationAluTest, LT)
 {
     EventEmitter<AluEvent> alu_event_emitter;
-    StrictMock<MockRangeCheck> range_check;
-    StrictMock<MockFieldGreaterThan> field_gt;
-    Alu alu(range_check, field_gt, alu_event_emitter);
+    StrictMock<MockGreaterThan> gt;
+    Alu alu(gt, alu_event_emitter);
 
     auto a = MemoryValue::from<uint32_t>(1);
     auto b = MemoryValue::from<uint32_t>(2);
 
-    EXPECT_CALL(range_check, assert_range(0, /*num_bits=*/32));
+    EXPECT_CALL(gt, gt(b, a));
```

Comment:
Since we now overload such that:
`gt(MemoryValue, MemoryValue)` --> `gt(uint128, uint128)`,
can/should we test the nested call gets to the latter ok? i.e. something like `EXPECT_CALL(gt, gt(uint128(b), uint128(a)));`

Reply:
I think if you are writing it like this (with the memoryvalues) the expect would be on the method with the memoryvalues. Since you are mocking, the inner redirection is really not something we'd need to test here.

You could test in the concrete GT object that
```
gt(MemoryValue, MemoryValue) --> gt(uint128, uint128),
```
However, this is easier in TS than in C++, since you have to mock either everything or nothing...

TL;DR: I guess from a behavioural PoV, this is enough for this test; and a test of the concrete GT class would have to test the 3 overloads separately without assuming they call each other (in theory).